infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java

```diff
@@ -28,11 +28,22 @@ public AbstractServer() {
         shutdown = false;
     }
 
-    /** Get the message handler for this instance.
-     * @return  A message handler.
+    /**
+     * Get the message handler for this instance.
+     *
+     * @return A message handler.
      */
     public abstract CorfuMsgHandler getHandler();
 
+    /**
+     * Seal the server with the epoch.
+     *
+     * @param epoch Epoch to seal with
+     */
+    public void sealServerWithEpoch(long epoch, IServerRouter r) {
```

Comment:
Can you get rid of the IServerRouter and just call it Seal ?

Reply:
Done.